common/cah-model.cpp

```diff
@@ -0,0 +1,196 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2020 Intel Corporation. All Rights Reserved.
+
+#include <algorithm>
+#include "cah-model.h"
+#define GLFW_INCLUDE_NONE
+#include "fw-update-helper.h"
+#include "model-views.h"
+
+using namespace rs2;
+
+// This registration should happen once in the SW runtime life.
+static bool registered_to_callback = false;
+
+// This variable is global for protecting the case when the callback will be called when the device model no longer exist.
+static std::atomic<rs2_calibration_status> calib_status = RS2_CALIBRATION_RETRY; 
+
+cah_model::cah_model() :_state(model_state_type::TRIGGER_MODAL),
+_process_started(false), _process_timeout()
+{
+    calib_status = RS2_CALIBRATION_RETRY;
```

Comment:
Why RETRY?

Reply:
It is the only suitable status I found that dont indicate start / finish of process
The enum is missing states like  "triggered", "idle", but I asked you and it's an API change so we did not touched it.
retry is the only enum value I can use with my logic.

Reply:
We'll revisit when auto trigger is done